src/Bounds.cpp

```diff
@@ -861,6 +965,44 @@ class Bounds : public IRVisitor {
                 // If the argument is unbounded on one side, then the max is unbounded.
                 interval.max = Interval::pos_inf;
             }
+        } else if (op->is_intrinsic(Call::absd)) {
+            internal_assert(!t.is_handle());
+            if (t.is_float()) {
+                Expr e = abs(op->args[0] - op->args[1]);
+                e.accept(this);
+            } else {
+                // absd() for int types will always produce a uint result
+                internal_assert(t.is_uint());
+
+                // Recover what the matched types for the args were
+                Expr a = op->args[0];
+                Expr b = op->args[1];
+                match_types(a, b);
+                internal_assert(a.type() == b.type());
+
+                a.accept(this);
+                Interval a_interval = interval;
+
+                b.accept(this);
+                Interval b_interval = interval;
+
+                a_interval.min = simplify(a_interval.min);
+                a_interval.max = simplify(a_interval.max);
+                b_interval.min = simplify(b_interval.min);
+                b_interval.max = simplify(b_interval.max);
+
+                if (a_interval.is_bounded() && b_interval.is_bounded()) {
+                    // Cast to 64-bit type to minimize cast-out-of-range issues;
```

Comment:
To avoid introducing wider types, how about:

interval.max = max(absd(a_interval.max, b_interval.min), absd(a_interval.min, b_interval.max))

Reply:
ah yes, but brings up another fun fact: `absd()` isn't handled in Simplify_Call.cpp, so `max(absd((uint8)20, (uint8)5), absd((uint8)10, (uint8)30))` isn't simplified down to `(uint8)20` as it should be, thus the newly added test fails

Reply:
(adding absd() handling to simplify now)